ecal-recon/src/main/java/org/hps/recon/ecal/cluster/ClusterPositionCorrection2021.java

```diff
@@ -0,0 +1,258 @@
+package org.hps.recon.ecal.cluster;
+
+import org.lcsim.event.base.BaseCluster;
+
+/**
+ * This uses the uncorrected cluster energy to correct the position of the cluster.
+ * This should be used before the energy is corrected on the Cluster and after
+ * cluster-track matching.
+ * This is to be used with 2021 data
+ *
+ * To determine these corrections, we simulated e+ e- and gamma at fixed
+ * energies over the ECAL acceptance, sampled the true hit position with MC
+ * scoring plane, and compared with the measured cluster position. We then
+ * considered:
+ * 
+ * dX vs X (dX = measured - true) ---> true=measured-dX dY vs Y (dY = measured -
+ * true) ---> true=measured-dY
+ * 
+ * We then performed a fit to these with dX = q + m*X dY = q1 + t*Y if x < 0 ; =
+ * q2 + t*Y if Y > 0
+ * 
+ * See also A.C. Talk at Nov. 2020 collaboration meeting
+ * 
+ * We then checked the dependency of the q,m, q1,q2, t parameters as a function
+ * of the energy
+ * 
+ * Electrons and Positrons: parameter(E) = p0 + p1*pow(E,p2) for all parameters
+ * Photons: par(E) = p0 + p1*pow(E,p2) | par = q,m par(E) = (a + b*E + c*E*E)/(d
+ * + e*E + f*E*E) | par = q1,t,q2
+ */
+final class ClusterPosResult21 {
+    private final double X;
+    private final double Y;
+
+    public ClusterPosResult21(double X, double Y) {
+        this.X = X;
+        this.Y = Y;
+    }
+
+    public double getX() {
+        return X;
+    }
+
+    public double getY() {
+        return Y;
+    }
+
+}
+
+public final class ClusterPositionCorrection2021 {
+    // Parameterizations tested in MC using v3-fieldmap
+    // Nov 2015
```

Comment:
We should use the database here rather than hard code numbers like this. Also, the code has a comment that these were tested on Nov 2015. I would like to see these tested them the 2021 beam energy and fieldmap.